src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -100,26 +59,146 @@ class ReactShallowRenderer {
-      createNodeMock: createShallowNodeMock(),
-    });
+    if (this._instance) {
+      this._rendered = updateClassComponent(
+        this._instance,
+        element.props,
+        context,
+      );
+    } else {
+      const prototype = element.type.prototype;
+
+      if (typeof prototype.render === 'function') {
+        this._instance = new element.type(
+          element.props,
+          context,
+          this._updater,
+        );
+
+        // TODO context validation: ReactDebugCurrentFrame
+        if (element.type.hasOwnProperty('contextTypes')) {
+          ReactDebugCurrentFrame.element = element;
+
+          checkPropTypes(
+            element.type.contextTypes,
+            context,
+            'context',
+            getName(element.type, this._instance),
+            ReactDebugCurrentFrame.getStackAddendum,
+          );
+
+          ReactDebugCurrentFrame.element = null;
+        }
+
+        this._rendered = mountClassComponent(
+          this._instance,
+          element.props,
+          context,
+        );
+      } else {
+        this._rendered = element.type();
+      }
+    }
 
     return this.getRenderOutput();
   }
 
   unmount() {
-    this._renderer.unmount();
+    if (this._instance) {
+      if (typeof this._instance.componentWillUnmount === 'function') {
+        this._instance.componentWillUnmount();
+      }
+    }
+
+    this._element = null;
+    this._rendered = null;
+    this._instance = null;
   }
 }
 
-// Backwards compatible API
-ReactShallowRenderer.createRenderer = function() {
-  return new ReactShallowRenderer();
-};
+class Updater {
+  constructor(renderer) {
+    this._renderer = renderer;
+  }
+
+  isMounted(publicInstance) {
+    return !!this._renderer._element;
+  }
+
+  enqueueForceUpdate(publicInstance, callback, callerName) {
+    this._renderer.render(this._renderer._element); // TODO
+  }
+
+  enqueueReplaceState(publicInstance, completeState, callback, callerName) {
+    publicInstance.state = completeState;
+    this._renderer.render(this._renderer._element);
+  }
+
+  enqueueSetState(publicInstance, partialState, callback, callerName) {
+    publicInstance.state = {
+      ...publicInstance.state,
+      ...partialState,
+    };
+    this._renderer.render(this._renderer._element);
+  }
+}
+
+function getName(type, instance) {
+  var constructor = instance && instance.constructor;
+  return type.displayName ||
+    (constructor && constructor.displayName) ||
+    type.name ||
+    (constructor && constructor.name) ||
+    null;
+}
+
+function mountClassComponent(
+  instance,
+  props = emptyObject,
+  context = emptyObject,
+) {
+  instance.context = context;
+  instance.props = props;
+  instance.state = instance.state || emptyObject;
+
+  if (typeof instance.componentWillMount === 'function') {
+    instance.componentWillMount();
+  }
+
+  const rendered = instance.render();
+
+  if (typeof instance.componentDidMount === 'function') {
+    instance.componentDidMount();
```

Comment:
Do we actually want to run this in shallow mode then? I would expect that this would crash if you use DOM refs. I don't think we used to fire it in 15.